theano/sandbox/gpuarray/__init__.py

```diff
@@ -53,10 +55,16 @@ def init_dev(dev):
         elif (config.init_gpu_device.startswith('cuda') or
               config.init_gpu_device.startswith('opencl')):
             init_dev(config.init_gpu_device)
+        if config.contexts != '':
+            for n, d in (c.split('->') for c in config.contexts.split(';')):
+                init_dev(d, n)
```

Comment:
call `strip()` on `d` and `n` to avoid spaces in the name of contexts or devices.